cmd/cli/telegram_utils.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/moira-alert/moira"
+	"github.com/moira-alert/moira/database/redis"
+	"github.com/moira-alert/moira/senders/telegram"
+)
+
+var (
+	telegramUsersKey = "moira-telegram-users:"
+	telegramLockName = "moira-telegram-users:moira-bot-host"
+)
+
+func updateTelegramUsersRecords(logger moira.Logger, database moira.Database) error {
+	logger.Info().Msg("Start updateTelegramUsersRecords")
+
+	switch d := database.(type) {
+	case *redis.DbConnector:
+		pipe := d.Client().TxPipeline()
+		iter := d.Client().Scan(d.Context(), 0, telegramUsersKey+"*", 0).Iterator()
+		for iter.Next(d.Context()) {
+			key := iter.Val()
+			if key == telegramLockName {
+				continue
+			}
+
+			oldValue, err := d.Client().Get(d.Context(), key).Result()
+			if err != nil {
+				return err
+			}
+
+			var chatID int64
+			var chat *telegram.Chat
+			var chatBytes []byte
+
+			chatID, err = strconv.ParseInt(oldValue, 10, 64)
+			if err != nil {
+				return err
+			}
+
+			if chatID < 0 {
+				chat = &telegram.Chat{
+					Type: "group",
+					ID:   chatID,
+				}
+			} else {
+				chat = &telegram.Chat{
+					Type: "private",
+					ID:   chatID,
+				}
+			}
+
+			chatBytes, err = json.Marshal(chat)
+			if err != nil {
+				return err
+			}
+
+			pipe.Set(d.Context(), key, string(chatBytes), 0)
+		}
+		_, err := pipe.Exec(d.Context())
+		if err != nil {
+			return err
+		}
+	default:
+		return makeUnknownDBError(database)
+	}
+
+	logger.Info().Msg("Successfully finished updateTelegramUsersRecords")
+
+	return nil
+}
+
+func downgradeTelegramUsersRecords(logger moira.Logger, database moira.Database) error {
+	logger.Info().Msg("Start downgradeTelegramUsersValue")
+
+	switch d := database.(type) {
+	case *redis.DbConnector:
+		pipe := d.Client().TxPipeline()
+		iter := d.Client().Scan(d.Context(), 0, telegramUsersKey+"*", 0).Iterator()
+		for iter.Next(d.Context()) {
+			key := iter.Val()
+			if key == telegramLockName {
```

Comment:
```go
var (
	telegramUsersKey = "moira-telegram-users:"
	telegramLockName = "moira-telegram-users:moira-bot-host:"
)

if strings.HasPrefix(key, telegramLockName) {
    continue
}
```

Reply:
Done in https://github.com/moira-alert/moira/commit/ff21de6760d0f4ca4df9f993a0288289d2b54a91